### ISSUE TITLE
parsing .- operator

### ISSUE THREAD

**Issue by noahyor** [disco-lang/disco#245]
2.-4 gives me negative 2, while 2 .- 4 gives me 0. why is that so?

**Comment by byorgey** [comment]
Hi @noahyor , thanks for reporting this!  What seems to be happening is that `2.-4` parses as `(2.) - 4`, whereas `2 .- 4` (with spaces) parses as one would expect. Notice how the first case has a `TRat` and a `Sub`, whereas the second has a `TNat` and `SSub`:
```
Disco> :parse 2.-4
TApp_ () (TPrim_ () (PrimBOp Sub)) (TTup_ () [TRat_ () (2 % 1),TNat_ () 4])
Disco> :parse 2 .- 4
TApp_ () (TPrim_ () (PrimBOp SSub)) (TTup_ () [TNat_ () 2,TNat_ () 4])
```
Although we don't have it written down anywhere as a formal guarantee, one might reasonably expect that putting space around operators is entirely optional, and it's weird in this case that you get different results depending on whether you insert spaces or not.

This [isn't the first time allowing a trailing period on numbers has gotten us in trouble](https://github.com/disco-lang/disco/issues/99). In that case `[1..]` was a parse error because it was parsing as `1.` followed by a single dot, instead of `1` followed by an ellipsis.

I thought allowing trailing periods as in `1.` was kind of cute at the time I added it, but (a) it's not actually standard mathematical notation, and (b) it seems to cause a lot of grief with parsing.  At this point I'm inclined to just disallow it (and require *e.g.* `1.0`).  We could also then get rid of [the parsing hack we used to solve #99](https://github.com/disco-lang/disco/commit/be7b51a7ecf2cb546a3ad3a3b39355ae108a0d62).
